image_writers_guide/guidelines.adoc

```diff
@@ -5,143 +5,186 @@
 
-This will make sure the next builds of the same image will be very fast because the cache will not be invalidated by upper layer changes.
+If instead you wrote the [sysitem]#Dockerfile# as:
 
-For example, if you are working on a Dockerfile that contains an `ADD` command to install a file you are iteratin on, and a `RUN` command to yum install a package, it's best to put the `ADD` command last, such as:
+----
+FROM foo
+ADD myfile /test/myfile
+RUN yum -y install mypackage && yum clean all -y
+----
 
-		FROM foo
-		RUN yum -y install mypackage && yum clean all -y
-		ADD myfile /test/myfile
+Then each time you changed `myfile` and reran `docker build`, the `ADD` operation would invalidate the `RUN` layer cache, so the `yum` operation would need to be rerun as well.
 
-This way each time you edit `myfile` and rerun `docker build` the system will reuse the cached layer for the yum command and only have to generate the new layer for the `ADD` operation.
+*Using `EXPOSE` to Mark Important Ports*
 
-If instead you wrote the Dockerfile as:
+See the "Always `EXPOSE` Important Ports" section of the http://www.projectatomic.io/docs/docker-image-author-guidance[Project Atomic documentation] for more information.
 
-		FROM foo
-		ADD myfile /test/myfile
-		RUN yum -y install mypackage && yum clean all -y
+*Setting Environment Variables with `ENV`*
 
-Then each time you changed `myfile` and reran `docker build`, the `ADD` operation would invalidate the `RUN` layer cache so the yum operation would need to be rerun as well.
+It is good practice to set environment variables with the `ENV` instruction. One example is to set the version of your project. This makes it easy for people find the version without looking at the [sysitem]#Dockerfile#. Another example is advertising a path on the system that could be used by another process, such as `JAVA_HOME`.
 
-=== Use EXPOSE to mark important ports
-See http://www.projectatomic.io/docs/docker-image-author-guidance/, subsection _Always `EXPOSE` Important Ports_
+*Avoiding Default Passwords*
 
-=== Set environment variables with ENV
-It’s a good idea to set environment variables with the `ENV` instruction. One example is to set version of your project. This will make it easy for people find the version without even looking at the Dockerfile.  Another example of an environment variable worth setting is when to advertise some path on the system that could be used by some process, like `JAVA_HOME`.
+Do not set default passwords. Many people will extend the image and forget to remove or change the default password. This can lead to security issues if there is a user with a well-known password in production. Passwords should be configurable using an environment variable instead. See the link:#using-env-vars[Using Environment Variables for Configuration] topic for more information.
```

Comment:
[Suggestion]
Do not set default passwords. ==>> It is best that you avoid setting default passwords. 

This ties it to the heading we are using. Starting a paragraph with "Do not...." sounded a bit rough.